File: lib/data/storage/migration/ile_migrator.py
class FileMigrator:
    def __init__(self, source_format, target_format):
        self.source_format = source_format
        self.target_format = target_format

    def migrate(self, lines):
        output = []
        for line in lines:
            data = self.source_format.deserialize_record(line)
            output.append(self.target_format.serialize_record(data))
        return output

File: lib/data/storage/migration/ile_migrator.dart
import 'dart:io';

import '../formats/record_format.dart';



class FileMigrator {
  final RecordFormat sourceFormat;
  final RecordFormat targetFormat;

  FileMigrator({required this.sourceFormat, required this.targetFormat});

  Future<void> migrate(String sourcePath, String targetPath) async {
    final raw = await File(sourcePath).readAsString();

    final records = sourceFormat.decode(raw);

    final converted = targetFormat.encode(records);

    await File(targetPath).writeAsString(converted);
  }
}
